services/app-api/src/postgres/contractAndRates/submitContract.ts

```diff
@@ -3,70 +3,68 @@ import type { ContractType } from '../../domain-models/contractAndRates'
 import { findContractWithHistory } from './findContractWithHistory'
 import { NotFoundError } from '../postgresErrors'
 import type { UpdateInfoType } from '../../domain-models'
-import { includeLatestSubmittedRateRev } from './prismaSubmittedRateHelpers'
+import { includeDraftRates } from './prismaDraftContractHelpers'
 
-type SubmitContractArgsType = {
+export type SubmitContractArgsType = {
     contractID: string // revision ID
     submittedByUserID: UpdateInfoType['updatedBy']
     submittedReason: UpdateInfoType['updatedReason']
 }
 // Update the given revision
 // * invalidate relationships of previous revision by marking as outdated
 // * set the UpdateInfo
-async function submitContract(
+export async function submitContract(
     client: PrismaClient,
     args: SubmitContractArgsType
 ): Promise<ContractType | NotFoundError | Error> {
     const { contractID, submittedByUserID, submittedReason } = args
     const currentDateTime = new Date()
 
     try {
-        // Find current contract revision with related rates
-        // query only the submitted revisions on the related rates
-        return await client.$transaction(async (tx) => {
-            const currentRev = await tx.contractRevisionTable.findFirst({
-                where: {
-                    contractID: contractID,
-                    submitInfoID: null,
-                },
-                include: {
-                    draftRates: {
-                        include: includeLatestSubmittedRateRev,
-                    },
+        // find the current contract with related rates
+        const currentRev = await client.contractRevisionTable.findFirst({
```

Comment:
why move this out of the transaction?

Reply:
Generally you don't need a read operation to be in a transaction because it has no bearing on rollback. Also it can make the transaction longer, which in our case is probably not a huge deal since we're low volume, but anything that can make a row or table lock for longer I tend to want to avoid. Since this one is easy to break out (as opposed to the later `findFirst`) I just pulled it to the top.

Reply:
I thought reading in a transaction would guarantee that you update based on the read data? 

Like if you read latestStateNumber as 0, then update it as 1, there's no chance that someone else would do the same in-between you?

Reply:
Oh yeah that's true, I just didn't see us using `currentRev` for that sort of data here. I'll move it back in.